apps/glusterfs/volume_entry.go

```diff
@@ -726,3 +726,43 @@ func eligibleClusters(db wdb.RODB, req ClusterReq,
 
 	return candidateClusters, err
 }
+
+func (v *VolumeEntry) runOnHost(db wdb.RODB,
+	cb func(host string) (bool, error)) error {
+
+	hosts := map[string]string{}
+	err := db.View(func(tx *bolt.Tx) error {
+		vol, err := NewVolumeEntryFromId(tx, v.Info.Id)
+		if err != nil {
+			return err
+		}
+
+		cluster, err := NewClusterEntryFromId(tx, vol.Info.Cluster)
+		if err != nil {
+			return err
+		}
+
+		for _, nodeId := range cluster.Info.Nodes {
+			node, err := NewNodeEntryFromId(tx, nodeId)
+			if err != nil {
+				return err
+			}
+			hosts[nodeId] = node.ManageHostName()
+		}
+
+		return nil
+	})
+	if err != nil {
+		logger.LogError("runOnHost failed to get hosts: %v", err)
+		return err
+	}
+
+	for nodeId, host := range hosts {
+		logger.Debug("running function on node %v (%v)", nodeId, host)
+		tryNext, err := cb(host)
+		if !tryNext {
```

Comment:
I would have found it a little more natural to do the logic roughly like this:
```
func (v *VolumeEntry) runOnHost(db wdb.RODB,
	cb func(host string) (error), bool tryNexy) error {
    ...
    for nodeId, host := range hosts {
        err := cb(host)
        if err == nil {
            return nil
        }
        if !tryNext {
            return err
        }
    }
    return fmt.Errorf(...)
}
```
Since I can't imagine the callback adapting the value of tryNext for different error cases. But maybe I'm just not seeing it. Won't block the merge for this. Functionally, it's perfectly fine as it is.